### DISCUSSION TITLE
Github: Skip archived repos at the earliest possible point (when listing repos)

### DISCUSSION THREAD

**Discussion by marcus-crane** [renovatebot/renovate#22042] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

Hi there,

We use a hosted instance of Renovate internally at my workplace and it's a really handy tool. We have a sizable number of archived repositories and I notice that Renovate (for Github anyway) only checks whether a repo is archived when it goes to process each one.

I think this could be handled much earlier, when listing repos (within the `getRepos` function in `lib/modules/platform/github/index.ts`). While each check is relatively quick (anywhere up to a second or two), it can still add a lot of processing time once you multiply that little bit of time across hundreds of dead repos.

I'm happy to submit a PR (and have already written an implementation locally) but it's suggested in the contributor guidelines that a discussion be opened first. It'd be nice to filter archives server side like Gitlab does but sadly it doesn't seem Github has such a feature (yet)

Thanks!

**Comment by rarkins** [comment]
Yes, that sounds like a good idea to me. Are you using an app token or PAT? Ideally both could be supported for consistency

**Reply by marcus-crane** [reply]
We're using an app token but my PR filters out archives for both setups. I'll open a PR shortly then 🙂

**Reply by marcus-crane** [reply]
https://github.com/renovatebot/renovate/pull/22057

**Reply by viceice** [reply]
Interestingly some platforms already do that filtering. We should also skip mirror repos and all other cases we can detect there
https://github.com/renovatebot/renovate/blob/ebdaa7eed5c3f6e7108d34e5fbc28c4f4d3b8a26/lib/modules/platform/gitlab/index.ts#L263-L289
